programs/stake/src/stake_state.rs

```diff
@@ -214,15 +217,47 @@ fn calculate_stake_points_and_credits(
     new_vote_state: &VoteState,
     stake_history: Option<&StakeHistory>,
     inflation_point_calc_tracer: Option<impl Fn(&InflationPointCalculationEvent)>,
-) -> (u128, u64) {
+    credits_auto_rewind: bool,
+) -> (u128, u64, bool) {
```

Comment:
Is it time to make this return type a struct?

Reply:
yeah, i tempted doing the cleanup. but, the tuple is still acceptable, imo...